### ISSUE TITLE
Missing Water Mesh (or Vector?) Piece

### ISSUE THREAD

**Issue by gerardsimons** [mapbox/mapbox-unity-sdk#398]
Hey guys, 

Here I am again. This time I have a problem with a water mesh not being generated properly, for tile `15/16786/10837` in the Rotterdam harbor area. As you can see from the image, all water is being generated properly except for the tile in the middle (although I tried changing the tile's relative position, which does not matter), where a piece of water is missing. I am using a modifier stack similar to the one for buildings in the `MeshGeneration.unity` example, with only minor changes in the height modifier (and a different `TextureModifier`). I tried different configurations, using flat tops or not, or toggling the force height option, still nothing.

I have included, some screenshots, showing the selected tile, the ModifierStack and a debugging view of visual studio at the `HeightModifier` for the tile, showing the contents of the `MeshData`, with what I think are too few vertices to show the missing part, resulting in me thinking the bug is probably inside the `MeshData` generator...? What do you guys think? It's a bit of mystery to me, at this point.

<img width="1440" alt="exhibit-a" src="https://user-images.githubusercontent.com/4247845/33242265-2d258ed6-d2d2-11e7-8334-15e9781c6e31.png">
<img width="1440" alt="exhibit-b" src="https://user-images.githubusercontent.com/4247845/33242267-30840788-d2d2-11e7-80eb-b9074717bbbf.png">
<img width="1440" alt="exhibit-c" src="https://user-images.githubusercontent.com/4247845/33242268-34eef90e-d2d2-11e7-8620-f8703acf8191.png">

**Comment by gerardsimons** [comment]
Actually I just found another tile (`15/16792/10837`) with a missing piece, very close to the above tile... Not sure if that is helpful information, but I thought it was worth sharing

<img width="1440" alt="screen shot 2017-11-26 at 18 08 29" src="https://user-images.githubusercontent.com/4247845/33242405-d7b31ccc-d2d4-11e7-861a-15d4bf1b1085.png">

**Comment by david-rhodes** [comment]
I validated the data in Mapbox Studio. Seems like this is an issue on our side. cc @brnkhy and ideas?

<img width="743" alt="screen shot 2017-11-27 at 10 17 14 am" src="https://user-images.githubusercontent.com/23202691/33280535-a874e3d8-d35f-11e7-90e3-19af70733809.png">

**Comment by brnkhy** [comment]
Hmm nice catch. I suspect it might be due to multiple disconnected parts. I'll check it 👍

**Comment by brnkhy** [comment]
This is due to tile clipping in the lower levels. 
![unity_2017-11-27_21-30-06](https://user-images.githubusercontent.com/1465390/33282604-2cba809e-d3ba-11e7-9a31-511df7755498.png)
as you can see, those two segments are actually connected out of the tile edges and one part gets lost during clipping process.  Actually yes, just found that we were only using the first item from the result of that clipping call. 

It worked when I changed "else" part in VectorTileFeature.cs line 90-110 to this;
```
List<List<Point2d<long>>> newGeom = new List<List<Point2d<long>>>();
					int geomCount = geom.Count;
					for (int i = 0; i < geomCount; i++)
					{
						List<Point2d<long>> part = geom[i];
						List<List<Point2d<long>>> tmp = new List<List<Point2d<long>>>();
						// flip order of inner rings to look like outer rings
						bool isInner = signedPolygonArea(part) >= 0;
						if (isInner) { part.Reverse(); }
						tmp.Add(part);
						tmp = UtilGeom.ClipGeometries(tmp, GeometryType, (long)_layer.Extent, clipBuffer.Value, scale.Value);
						// ring was completely outside of clip border
						if (tmp.Count == 0)
						{
							continue;
						}

						foreach (var item in tmp)
						{
							// flip winding order of inner rings back
							if (isInner) { item.Reverse(); }
							newGeom.Add(item);
						}
					}
					geom = newGeom;
```
but I'm not sure if that'll break something else (couldn't test much as I'm having connection issues).

cc @BergWerkGIS

**Comment by brnkhy** [comment]
![unity_2017-11-27_21-38-26](https://user-images.githubusercontent.com/1465390/33282948-51487b40-d3bb-11e7-8825-6241ebd4c03b.png)
after change

**Comment by gerardsimons** [comment]
Awesome! Thanks a lot guys, that helped me out with my problem, at least, let's hope that it doesn't break anything else. Cheers

**Comment by brnkhy** [comment]
Glad it worked! Let's keep this open until we fix and merge the bug though.

**Comment by BergWerkGIS** [comment]
fixed in https://github.com/mapbox/vector-tile-cs/commit/59eb9264a9cae52b1d5b7c47e51b9231467dea6e

The fix will be part of the upcoming v1.3.0 release of the SDK.
